src/GuestConfiguration/GuestConfiguration/Common/GuestConfigurationCmdletBase.cs

```diff
@@ -310,7 +322,16 @@ private IEnumerable<PolicyData> GetPolicyStatusesHelper(PolicySetDefinition[] po
                 foreach (var policyDefinitionIdInInitiative in policyDefinitionIdsInInitiative)
                 {
                     var _initiativeName = GetInitiativeNameFromId(policyDefinitionIdInInitiative);
-                    var policyDef = PolicyClient.PolicyDefinitions.GetBuiltIn(_initiativeName);
+                    PolicyDefinition policyDef;
+                    try
+                    {
+                        policyDef = PolicyClient.PolicyDefinitions.GetBuiltIn(_initiativeName);
```

Comment:
how do you distinguish between service error vs builtin/custom policy ?

Reply:
catching specific exception and adding when clause corresponding to not found status code.

---
In reply to: [258725693](https://github.com/Azure/azure-powershell/pull/8574#discussion_r258725693) [](ancestors = 258725693)

Reply:
have you tested this works for both custom and builtin policies ?